File: project/jooeun-onboarding/survey-api/src/main/java/com/innercircle/survey/api/exception/SurveyNotFoundException.java
package com.innercircle.survey.api.exception;

import com.innercircle.survey.common.exception.BusinessException;
import com.innercircle.survey.common.exception.ErrorCode;

import java.util.Map;

/**
 * 설문조사를 찾을 수 없을 때 발생하는 예외
 */
public class SurveyNotFoundException extends BusinessException {
    
    public SurveyNotFoundException(String surveyId) {
        super(ErrorCode.SURVEY_NOT_FOUND, 
              String.format("설문조사를 찾을 수 없습니다: %s", surveyId),
              Map.of("surveyId", surveyId));
    }
}

File: project/kyungrok-onboarding/src/main/kotlin/kr/innercircle/onboarding/survey/exception/SurveyNotFoundException.kt
package kr.innercircle.onboarding.survey.exception

import kr.innercircle.onboarding.survey.exception.base.BadRequestException
import kr.innercircle.onboarding.survey.exception.base.ErrorCode

/**
 * packageName : kr.innercircle.onboarding.survey.exception
 * fileName    : SurveyNotFoundException
 * author      : ckr
 * date        : 25. 6. 21.
 * description :
 */
class SurveyNotFoundException: BadRequestException(ErrorCode.SURVEY_NOT_FOUND)